### PR TITLE
TRT-786: Relax the per-job disruption tests and fix bugs/improve consistency

### PR DESCRIPTION
**Pull request by dgoodwin**
Details for the why available in the linked jira.

- Allow 5s or 20% grace on per-job disruption testing
- Drop all disruption/alert matcher fallbacks except previous release
- Stop disruption testing when platform is unknown
- Fix broken new/reused disruption test names causing inaccurate flakes
- Update image registry and PDB disruption test names for consistency

### CODE REVIEW

**Comment by dgoodwin** [comment]
I think this is ready for review. Here is what the output looks like when it fails (forced via a hack earlier in the PR runs to always fail):

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28453/pull-ci-openshift-origin-master-e2e-gcp-ovn-upgrade/1734281678186090496

File: pkg/monitortestlibrary/disruptionlibrary/disruption_invariant_adapter.go [hunk 1]
```diff
@@ -98,7 +99,24 @@ func (w *Availability) CollectData(ctx context.Context) (monitorapi.Intervals, [
...
+
+	// Not sure what these are, but this will help find them, and we don't get any value from testing these:
+	if jobType.Platform == "" {
+		return &junitapi.JUnitTestCase{
+			Name: testName,
+			SkipMessage: &junitapi.SkipMessage{
+				Message: "Unknown platform, skipping disruption testing",
+			},
+		}
+	}
```
Comment (stbenjam) [inline pkg/monitortestlibrary/disruptionlibrary/disruption_invariant_adapter.go:118]: Would it make sense to make this a notSupportedReason instead?

Comment (stbenjam) [reply pkg/monitortestlibrary/disruptionlibrary/disruption_invariant_adapter.go:118]: Sure that sounds fine if you want to have the data

**Pull request merged**